Dockerfile

```diff
@@ -5,11 +5,28 @@ FROM python:3.8
 COPY --from=thrift /usr/local/bin/thrift /usr/local/bin/thrift
 COPY --from=go /usr/local/go /usr/local/go
 
-ENV GOPATH=/root/go
+RUN mkdir /opt/go
+ENV GOPATH=/opt/go
 ENV PATH=$GOPATH/bin:/usr/local/go/bin:$PATH
 RUN go install golang.org/x/lint/golint@latest
 
 WORKDIR /src
 COPY lib/py/requirements*.txt ./
 RUN pip install -r requirements.txt
+
+# Permission fixes for running as a test environment.
+# Assumption: This docker image is not used outside build and test
+# environments.
+# $GOPATH: conflicts with local volume mount of /src:
+#    This mount provides a live code editing environment.
+#    It is not accessed as root (must use the host's UID namespace to write
+#    back to the volume) so to have a single process reading $GOPATH and /src
+#    simultaneously we need open permissions.
+# $HOME/.cache:
+#    go expects $HOME/.cache to be writable but in the recommended test
+#    environment, $HOME is /, so pre-create .cache too.
+
+RUN mkdir /.cache
+RUN chmod a+rwX -R $GOPATH /.cache
```

Comment:
🔕 the original `$GOPATH` (that is `/root/go`) should already be writable, otherwise the `go install` line (line 10 in old code and 11 in new code) would fail (as it also writes to `$GOPATH`), so I'm not sure what problem you hit with `$GOPATH`.

Reply:
It's writable as root, but the local test instructions have you running as a different UID to provide live edit access to the host filesystem:  `docker run -it -v $PWD:/src --user "$(id -u):$(id -g)" -w /src edgecontext`.

Now that the tests work, I can certainly back this out, but it's a bit funny.

Reply:
oh right you are trying to map the directory to your host machine and do local development. that makes sense.